test/sim/moves/terablast.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+describe('Tera Blast', function () {
+	it(`should be a special attack when base stats are tied`, function () {
+		const battle = common.createBattle([[
+			// Regidrago has equal base attack and special attack stats.
+			{species: 'regidrago', ability: 'dragonsmaw', moves: ['terablast']},
+		], [
+			{species: 'regirock', ability: 'clearbody', moves: ['protect']},
+		]]);
+		battle.makeChoices('move terablast terastallize', 'move protect');
+
+		const leftPokemon = battle.p1.pokemon[0];
+		assert.notEqual(leftPokemon.terastallized, undefined);
```

Comment:
It's not really necessary to check whether or not the Pokemon is Terastallized, that's pretty trivial. I would remove these checks.

Reply:
Good point. Removed!